### PR TITLE
WIP Improve tests for Ownable, add tests for PullPaymentBid, consider returning a boolean on bid()

### PR DESCRIPTION
**Pull request by adklempner**
PullPaymentBid.bid() currently throws an error if a bid is lower than the highest bid. 

```
Error: VM Exception while executing transaction: invalid JUMP
```

Is there a specific reason for this? Perhaps it would be better to return a boolean indicating whether the bid was successful or not.

### CODE REVIEW

**Comment by maraoz** [comment]
As a general rule for contributing: don't build your changes over other unmerged PRs. This PR has changes corresponding to https://github.com/OpenZeppelin/zep-solidity/pull/20/.
It's better to create separate branches from master so that PRs are more clear and easy to review.

File: contracts/examples/PullPaymentBid.sol [hunk 1]
```diff
@@ -3,14 +3,15 @@ import '../PullPaymentCapable.sol';
...
   address public highestBidder;
   uint public highestBid;
-  
-  function bid() external {
-    if (msg.value <= highestBid) throw;
-    
+
+  function bid() external returns (bool) {
+    if (msg.value <= highestBid) return false;
+
```
Comment (maraoz) [inline contracts/examples/PullPaymentBid.sol]: Returning false here makes `msg.value` to be added to the contract balance, but no bid is placed! This breaks the contract logic. That's why throw is required here

**Comment by adklempner** [comment]
Ah, I see. Thanks for the clarification.

**Comment by maraoz** [comment]
Needs rebase and fixing based on comments

**Pull request closed without merging**